lib/search.bash

```diff
@@ -58,137 +185,261 @@ _bash-it-array-contains-element () {
   echo -n $r
 }
 
-_bash-it-search-component() {
-  _about 'searches for given terms amongst a given component'
-  _param '1: component type, one of: [ aliases | plugins | completions ]'
-  _param '2: term1 '
-  _param '3: [-]term2 [-]term3 ...'
-  _example '$ _bash-it-search-component aliases rake bundler -chruby'
+_bash-it-array-dedup() {
+  echo "$*" | tr ' ' '\n' | sort -u | tr '\n' ' '
+}
+
+_bash-it-grep() {
```

Comment:
I wasn’t sure what the policy was, but at least this one, array include test and array deed are totally generic functions.

Another topic is that the search cache can probably be used just as well by the main library. Write operations (enable/disable) perform delayed cache  expiration, so anytime someone tries to eg show all aliases, we could possibly read it from the cache this speeding up this seemingly long operation. But that would be another PR if you think it’s a good idea.

Reply:
I wasn’t sure what the policy was, but at least this one, array include test and array deed are totally generic functions.

Another topic is that the search cache can probably be used just as well by the main library. Write operations (enable/disable) perform delayed cache  expiration, so anytime someone tries to eg show all aliases, we could possibly read it from the cache this speeding up this seemingly long operation. But that would be another PR if you think it’s a good idea.

Reply:
Great! I extracted common functions to utilities.bash — did not want to add more to helpers, since it's already quite large.

I also removed the somewhat fragile cache cleanup deferral code that was speeding up `--enable/--disable` on a large search result until I have a more reliable version.